rs/types/base_types/src/canister_id.rs

```diff
@@ -187,14 +233,12 @@ impl<'de> Deserialize<'de> for CanisterId {
     where
         D: serde::de::Deserializer<'de>,
     {
-        // Not all principals are valid inside a CanisterId.
-        // Therefore, deserialization must explicitly
-        // transform the PrincipalId into a CanisterId.
-        // A derived implementation of Deserialize would open
-        // the door to invariant violation.
-        let res = CanisterId::try_from(PrincipalId::deserialize(deserializer)?);
-        let id = res.map_err(D::Error::custom)?;
-        Ok(id)
+        let result = PrincipalId::deserialize(deserializer)?;
+        // TODO: Ideally, instead of unchecked_from_principal, we'd use
+        // try_from_principal_id, but that would break people who are relying on
+        // the original behavior of CanisterId::try_from(principal_id), which
+        // would ALWAYS return Ok.
+        Ok(CanisterId::unchecked_from_principal(result))
```

Comment:
Similar comment as above. If we plan to change `try_from` eventually, then we might want to wait until that PR to touch this here?

Reply:
Done.